var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -674,6 +674,24 @@ def symlink_to_h5hl_wrappers(self):
                         os.remove(old)
                         symlink(new, old)
 
+    @run_after("install")
+    def symlink_mpi_libs(self):
+        """Compatibility layer to support projects looking for the MPI suffix"""
+        if not self.spec.satisfies("+mpi"):
+            return
+
+        with working_dir(self.prefix.lib):
+            symlink_libs = [
+                "libhdf5{mpi_suffix}{lib_suffix}",
+                "libhdf5{mpi_suffix}_hl{lib_suffix}",
+            ]
+            for lib_f in symlink_libs:
+                for lib_suffix in (".so", ".a"):
```

Comment:
What about macOS? Do we need to use `dso_suffix`?

Reply:
Good call, updated to use `find_libraries` instead to capture this correctly on all systems.